### PR TITLE
make ceph work

### PR DESCRIPTION
**Pull request by durandom**
implement ceph storage api similar to s3

@tumido please review

### CODE REVIEW

File: storage/ceph.py [hunk 1]
```diff
@@ -1,14 +1,54 @@
 import pyarrow.parquet as pq
-import s3fs
+import re
+import logging
+import pickle
+import s3fs, datetime, os
```
Comment (tumido) [inline storage/ceph.py]: If you'd be interested in nice Python, I suggest implementing (using) PEP8. :wink: 

For example here, [about imports](https://www.python.org/dev/peps/pep-0008/#imports) it says:
- Imports should usually be on separate lines
- Imports should be grouped in the following order: Standard library imports, Related third party imports, Local application/library specific imports

File: storage/ceph.py [hunk 2]
```diff
@@ -1,14 +1,54 @@
...
+                if re.match(r"\d{4}-\d{2}-\d{2}", date):
+                    cache.update({date: p})
+    except FileNotFoundError:
+        cache = {}
+
+    return cache
+
+def unprocessed():
+    processed = set(available())
+    start = datetime.datetime(2018, 6, 1)
```
[the remark this answers is not in the document]
Comment (durandom) [reply storage/ceph.py]: black magic. yes :)
I'd leave it as this for now

**Comment by durandom** [comment]
Actually the code really is basically the S3 code. In a follow up we should make the ceph code use the s3 stuff directly. Or use proper inheritance and not this conditional import hack?!

But this should do for now

File: storage/s3.py [hunk 3]
```diff
@@ -14,9 +14,12 @@
 def available(cache={}):
     try:
         if not cache:
+            print(DEST_BUCKET)
+            print(fs.ls(DEST_BUCKET))
```
Comment (tumido) [inline storage/s3.py]: I think you've forgotten this here.

File: storage/s3.py [hunk 4]
```diff
@@ -14,9 +14,12 @@
...
             cache.update({os.path.basename(p).split(".")[0]: p for p in fs.ls(DEST_BUCKET)})
     except FileNotFoundError:
         cache = {}
+    print(cache)
```
Comment (tumido) [inline storage/s3.py]: ... and neither this print should be here.

**Comment by tumido** [comment]
Yes, maybe implementing this as classes instead of plain modules can help. But.. that's not in scope of this PR for sure. :wink:

File: storage/s3.py (hunk 4 above)
Comment (durandom) [reply storage/s3.py]: oooh. those darn debugging prints...

**Comment by durandom** [comment]
@tumido thanks for the review. merge now?

**Comment by durandom** [comment]
@tumido you should be able to merge now

**Pull request merged**